reactos/drivers/usb/usbmport.h

```diff
@@ -0,0 +1,716 @@
+  PHCI_RH_GET_HUB_STATUS RH_GetHubStatus;
+  PHCI_RH_SET_FEATURE_PORT_RESET RH_SetFeaturePortReset;
+  PHCI_RH_SET_FEATURE_PORT_POWER RH_SetFeaturePortPower;
+  PHCI_RH_SET_FEATURE_PORT_ENABLE RH_SetFeaturePortEnable;
+  PHCI_RH_SET_FEATURE_PORT_SUSPEND RH_SetFeaturePortSuspend;
+  PHCI_RH_CLEAR_FEATURE_PORT_ENABLE RH_ClearFeaturePortEnable;
+  PHCI_RH_CLEAR_FEATURE_PORT_POWER RH_ClearFeaturePortPower;
+  PHCI_RH_CLEAR_FEATURE_PORT_SUSPEND RH_ClearFeaturePortSuspend;
+  PHCI_RH_CLEAR_FEATURE_PORT_ENABLE_CHANGE RH_ClearFeaturePortEnableChange;
+  PHCI_RH_CLEAR_FEATURE_PORT_CONNECT_CHANGE RH_ClearFeaturePortConnectChange;
+  PHCI_RH_CLEAR_FEATURE_PORT_RESET_CHANGE RH_ClearFeaturePortResetChange;
+  PHCI_RH_CLEAR_FEATURE_PORT_SUSPEND_CHANGE RH_ClearFeaturePortSuspendChange;
+  PHCI_RH_CLEAR_FEATURE_PORT_OVERCURRENT_CHANGE RH_ClearFeaturePortOvercurrentChange;
+  PHCI_RH_DISABLE_IRQ RH_DisableIrq;
+  PHCI_RH_ENABLE_IRQ RH_EnableIrq;
+
+  /* Miniport ioctl */
+
+  PHCI_START_SEND_ONE_PACKET StartSendOnePacket;
+  PHCI_END_SEND_ONE_PACKET EndSendOnePacket;
+  PHCI_PASS_THRU PassThru;
+
+  /* Port */
+
+  PUSBPORT_DBG_PRINT UsbPortDbgPrint;
+  PUSBPORT_TEST_DEBUG_BREAK UsbPortTestDebugBreak;
+  PUSBPORT_ASSERT_FAILURE UsbPortAssertFailure;
+  PUSBPORT_GET_MINIPORT_REGISTRY_KEY_VALUE UsbPortGetMiniportRegistryKeyValue;
+  PUSBPORT_INVALIDATE_ROOT_HUB UsbPortInvalidateRootHub;
+  PUSBPORT_INVALIDATE_ENDPOINT UsbPortInvalidateEndpoint;
+  PUSBPORT_COMPLETE_TRANSFER UsbPortCompleteTransfer;
+  PUSBPORT_COMPLETE_ISO_TRANSFER UsbPortCompleteIsoTransfer;
+  PUSBPORT_LOG_ENTRY UsbPortLogEntry;
+  PUSBPORT_GET_MAPPED_VIRTUAL_ADDRESS UsbPortGetMappedVirtualAddress;
+  PUSBPORT_REQUEST_ASYNC_CALLBACK UsbPortRequestAsyncCallback;
+  PUSBPORT_READ_WRITE_CONFIG_SPACE UsbPortReadWriteConfigSpace;
+  PUSBPORT_WAIT UsbPortWait;
+  PUSBPORT_INVALIDATE_CONTROLLER UsbPortInvalidateController;
+  PUSBPORT_BUG_CHECK UsbPortBugCheck;
+  PUSBPORT_NOTIFY_DOUBLE_BUFFER UsbPortNotifyDoubleBuffer;
+
+  /* Miniport */
+
+  PHCI_REBALANCE_ENDPOINT RebalanceEndpoint;
+  PHCI_FLUSH_INTERRUPTS FlushInterrupts;
+  PHCI_RH_CHIRP_ROOT_PORT RH_ChirpRootPort;
+  PHCI_TAKE_PORT_CONTROL TakePortControl;
+  ULONG Reserved4;
+  ULONG Reserved5;
+} USBPORT_REGISTRATION_PACKET, *PUSBPORT_REGISTRATION_PACKET;
+
+typedef struct _USBPORT_MINIPORT_INTERFACE {
+  PDRIVER_OBJECT DriverObject;
+  LIST_ENTRY DriverLink;
+  PDRIVER_UNLOAD DriverUnload;
+  ULONG Version;
+  USBPORT_REGISTRATION_PACKET Packet;
+} USBPORT_MINIPORT_INTERFACE, *PUSBPORT_MINIPORT_INTERFACE;
+
+C_ASSERT(sizeof(USBPORT_MINIPORT_INTERFACE) == 336);
+
+typedef struct _USBPORT_ENDPOINT_PROPERTIES {
+  USHORT DeviceAddress;
+  USHORT EndpointAddress;
+  USHORT TotalMaxPacketSize; // TransactionPerMicroframe * MaxPacketSize
+  UCHAR Period;
+  UCHAR Reserved1;
+  USB_DEVICE_SPEED DeviceSpeed;
+  ULONG UsbBandwidth;
+  ULONG ScheduleOffset;
+  ULONG TransferType;
+  ULONG Direction;
+  ULONG BufferVA;
+  ULONG BufferPA;
```

Comment:
Changed to ULONG_PTR. This is the USBPORT_COMMON_BUFFER_HEADER::PhysicalAddress == LogicalAddress.LowPart & 0xFFFFF000